Feature/ae 1173 ohjesivu

muep opened the pull request.

muep commented:
Tätä saa jo kommentoida jos haluaa ja varsinkin mikäli näyttää että on menossa ihan metsään. Tässä vaiheessa ainakin tuo update-osasto on vielä menossa uusiksi.

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/service/sivu_test.clj:
```diff
@@ -0,0 +1,53 @@
+    (t/is (:published sivu-out))))
+
+(t/deftest simple-hierarchy
+  (let [whoami {:rooli 2 :id 1}
```
testi-data-setin käyttö voisi tehdä itse testit lyhyemmiksi. Mutta vähän makuasia.

muep replied:
Katsotaanpa, tähän asiaan ei vielä tullut paljoa perehdyttyä, mutta ehkä olisi syytä.

muep replied:
Nyt äskettäin pushatussa versiossa on generoitua testidataa

muep commented:
Voidaan lopuksi squashata tämä yhdeksi commitiksi, mutta minusta kannattaa pitää muutokset erillään sen aikaa kun review on kesken, niin on eri reviewkerroilla helppo nähdä että mitä on muuttunut edellisen vilkaisun jälkeen.

muep commented on etp-backend/src/test/clj/solita/etp/service/laatija_test.clj:
```diff
@@ -177,10 +177,10 @@
-                #(service/validate-laatija-patevyys! ts/*db* id))
-               {:type          :patevyys-expired
-                :paattymisaika (patevyys-paattymisaika expired-3-years-ago)}))
+      (comment (t/is (= (etp-test/catch-ex-data-no-msg
```
Tämä ei taida kuulua tänne

muep committed:
AE-1173 Minimal ohjesivu backend

Web API for the following operations:
- add a sivu
- get metadata of all sivut
- get a single sivu entirely
- update fields of a sivu

In addition to AE-1173, this also addresses:
- AE-1155 Sivun muokkaminen - backend
- AE-1161 Sivuhierarkian hakeminen - backend
- AE-1165 Ohjesivujen tietomalli ja tietokantamigraatiot

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/service/sivu_test.clj:
```diff
@@ -0,0 +1,135 @@
+     :unpublished-sivut unpublished-sivut}))
+
+(t/deftest find-sivu
+  (let [ds (test-data-set)
```
Tässä voi suoraan deskturoida, jos haluaa.

muep replied:
Minusta tässä tilanteessa ei tunnu kannattavalta, kun noita kahta sisältä otettavaa käytetään kuitenkin vain kerran, ja sisältä otettavat asiat on helppo seurata noista threading macroista.

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/service/sivu_test.clj:
```diff
@@ -0,0 +1,135 @@
+
+(t/deftest find-sivu
+  (let [ds (test-data-set)
+        sivut-by-id (reduce #(assoc %1 (:id %2) %2) {} (:sivut ds))
```
Tässä toimii kivasti group-by reducen sijasta.

muep replied:
Vaihdettu

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/service/sivu_test.clj:
```diff
@@ -0,0 +1,135 @@
+(t/deftest find-sivu
+  (let [ds (test-data-set)
+        sivut-by-id (reduce #(assoc %1 (:id %2) %2) {} (:sivut ds))
+        laatija {:rooli 0}
```
kayttaja-test-data -nimiavaruudesta löytyy laatija ja paakayttaja -helpperit suoraan, millä saa testiä vähän lyhyemmäksi.

muep replied:
laitoin paikalleen

MattiNieminen commented on etp-backend/src/test/clj/solita/etp/service/sivu_test.clj:
```diff
@@ -0,0 +1,135 @@
+    (t/is (= (:body  sivu-out) (:body  sivu-in)))
+    (t/is (:published sivu-out))))
+
+(t/deftest simple-hierarchy
```
Tätä testiä saisi jonkin verran lyhyemmäksi, jos generoisi nuo insertoitavat sivut test-data-generatorilla tai suoraan test-data-setissä. Titlen ja bodynhan ei tarvitse olla deterministisiä, kunhan niihin on bindingit ja niihin voidaan vertailla.

muep replied:
Mietin että ottaisiko tämän vain pois, mutta minusta se on enimmäkseen hyödyllinen helpommin ymmärrettävänä testinä siinä tilanteessa, että joku muutos särkee laajasti. Jos tälle piirteelle ei ole tarvetta, niin tämä ei kovin paljoa muuten ehkä hyödytä pitää ylipäänsä mukana.

muep committed:
AE-1173 group-by instead of reduce

muep committed:
AE-1173 Use roles from kayttaja-test-data
